File: Java/Tema2_Ejercicio9.java
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 9
 * ================================================
 * Escriba un programa que calcule potencias usando operadores.
 * 
 * EXPLICACIÓN:
 * - Usamos Math.pow(base, exponente) para calcular potencias
 * - También podemos usar un bucle para multiplicar repetidamente
 * - Demostramos ambas formas
 */

import java.util.Scanner;

public class Tema2_Ejercicio9 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite la base: ");
        int base = scanner.nextInt();
        
        System.out.print("Digite el exponente: ");
        int exponente = scanner.nextInt();
        
        // Forma 1: Usando Math.pow()
        double resultadoMath = Math.pow(base, exponente);
        
        // Forma 2: Usando bucle
        long resultadoBucle = 1;
        for (int i = 0; i < exponente; i++) {
            resultadoBucle *= base;
        }
        
        System.out.println("\n=== CÁLCULO DE POTENCIA ===");
        System.out.println("Base: " + base);
        System.out.println("Exponente: " + exponente);
        System.out.println("Resultado (Math.pow()): " + (long)resultadoMath);
        System.out.println("Resultado (usando bucle): " + resultadoBucle);
        
        // Mostrar el cálculo paso a paso
        System.out.println("\nPasos:");
        long paso = 1;
        for (int i = 1; i <= exponente; i++) {
            paso *= base;
            System.out.println("  " + base + "^" + i + " = " + paso);
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * ================
 * Digite la base: 2
 * Digite el exponente: 5
 * 
 * === CÁLCULO DE POTENCIA ===
 * Base: 2
 * Exponente: 5
 * Resultado (Math.pow()): 32
 * Resultado (usando bucle): 32
 * 
 * Pasos:
 *   2^1 = 2
 *   2^2 = 4
 *   2^3 = 8
 *   2^4 = 16
 *   2^5 = 32
 */

File: JS/Tema2_Ejercicio9.js
/*
 * TEMA 2: EXPRESIONES Y OPERADORES - EJERCICIO 9
 * ===============================================
 * Calcular potencias usando el operador **.
 * 
 * EXPLICACIÓN:
 * - Operador ** para potenciación
 * - Equivalente a Math.pow()
 * - Exponentes positivos y negativos
 */

function calcularPotencias(base, exponente) {
    const resultado = base ** exponente;
    return resultado;
}

console.log("=== CÁLCULO DE POTENCIAS ===");

console.log("Potencias positivas:");
console.log(`2^3 = ${calcularPotencias(2, 3)}`);
console.log(`5^2 = ${calcularPotencias(5, 2)}`);
console.log(`10^3 = ${calcularPotencias(10, 3)}`);

console.log("\nPotencias negativas (inverso):");
console.log(`2^-1 = ${calcularPotencias(2, -1)}`);
console.log(`5^-2 = ${calcularPotencias(5, -2)}`);

console.log("\nPotencias fraccionarias (raíces):");
console.log(`16^(1/2) = ${calcularPotencias(16, 1/2)}`);  // Raíz cuadrada
console.log(`27^(1/3) = ${calcularPotencias(27, 1/3)}`);  // Raíz cúbica

console.log("\nComparación con Math.pow():");
console.log(`Math.pow(2, 3) = ${Math.pow(2, 3)}`);
console.log(`2 ** 3 = ${2 ** 3}`);

/*
 * SALIDA ESPERADA:
 * === CÁLCULO DE POTENCIAS ===
 * Potencias positivas:
 * 2^3 = 8
 * 5^2 = 25
 * 10^3 = 1000
 * 
 * Potencias negativas (inverso):
 * 2^-1 = 0.5
 * 5^-2 = 0.04
 * 
 * Potencias fraccionarias (raíces):
 * 16^(1/2) = 4
 * 27^(1/3) = 3
 * 
 * Comparación con Math.pow():
 * Math.pow(2, 3) = 8
 * 2 ** 3 = 8
 */
